exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -173,6 +174,7 @@ func newMetricsExporter(config Config, set component.ExporterCreateSettings, mar
 	if marshaler == nil {
 		return nil, errUnrecognizedEncoding
 	}
+	set.Logger.Info(fmt.Sprintf(logMarshalerUsedMsg, marshaler))
```

Comment:
What is the intent of these log lines?

I would encourage the use of structured logging instead of making a called to fmt.Sprintf

Reply:
```suggestion
set.Logger.Debug("Configured marshaller", zap.String("encoding", config.Encoding))
```

Reply:
The strategy for the test is to validate the message contents via a zap.Hook.  Unfortunately, the fields of the message are not provided in the zapcore.Entry provided to the hook.

Reply:
I think @MovieStoreGuy is right. Are you sure about "Unfortunately, the fields of the message are not provided in the zapcore.Entry provided to the hook.", does seem like a bug in zap, and hard to believe it.

Reply:
Oh interesting, I wasn't aware that logs lines were covered as part of the test.

From my perspective, I don't see too much value in validating that log lines happened since it often leads into an antipattern of checking if logs exist to see if some functionality has been reached. There should be other state to be able to inspect to validate that it has indeed done what it says it has.


Furthermore, These are more just feedback to the user who provided the configuration. They may or may not find them useful.

Reply:
As per my comment on structured logging, the reason I'm firm for this is that it makes search logs more efficient in something like loki, splunk, tail -f | grep, pick your poison. 

Keeping the message field static means that I can take that text and either include or exclude those lines really easily and those systems can take advantage of their engines that can optimise on that static string. (From my understand, some use bloom filters to make filtering easier). 

Finally, not that it is strictly important here but it does remove one additional allocation for the string which may not seem like much but it definitely adds up when you consider all the components and core service of the collector.